### PR TITLE
Fix #2476 Kudu should recover from Corrupted SiteExtension settings

### LINKED ISSUE #2476: Kudu should recover from Corrupted SiteExtension settings
This setting file can get corrupted if site restarted in the middle.   Kudu should recover from it.

```csharp
Error occurred type="error" text="Error reading JObject from JsonReader. Path '', line 0, position 0." stackTrace="
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Kudu.Core.Settings.JsonSettings.<Read>b__12_0()
   at Kudu.Contracts.Infrastructure.LockExtensions.<>c__DisplayClass3_0`1.<LockOperation>b__0()
   at Kudu.Contracts.Infrastructure.LockExtensions.TryLockOperation(IOperationLock lockObj, Action operation, String operationName, TimeSpan timeout)
   at Kudu.Contracts.Infrastructure.LockExtensions.LockOperation[T](IOperationLock lockObj, Func`1 operation, String operationName, TimeSpan timeout)
   at Kudu.Core.Settings.JsonSettings.Read()
   at Kudu.Core.Settings.JsonSettings.GetValue(String key)
```

### PR DESCRIPTION
**Pull request by suwatch**

### CODE REVIEW

File: Kudu.Services/SiteExtensions/SiteExtensionController.cs [hunk 1]
```diff
@@ -233,14 +233,9 @@ public SiteExtensionController(ISiteExtensionManager manager, IEnvironment envir
...
 
-                if (tracer.TraceLevel == TraceLevel.Off)
-                {
-                    backgroundTracer = NullTracer.Instance;
-                }
-
                 if (tracer.TraceLevel > TraceLevel.Off)
                 {
-                    backgroundTracer = new XmlTracer(_environment.TracePath, tracer.TraceLevel);
+                    backgroundTracer = new CascadeTracer(new XmlTracer(_environment.TracePath, tracer.TraceLevel), new ETWTracer(_environment.RequestId, "PUT"));
```
Comment (davidebbo) [inline Kudu.Services/SiteExtensions/SiteExtensionController.cs]: Did this intend to be in same CR? Seems unrelated.

Comment (suwatch) [inline Kudu.Services/SiteExtensions/SiteExtensionController.cs]: Unrelated.  When testing it, I noticed no ETW trace when we go async SiteExtension installation in BGThread.   I can break into 2 commits.

Comment (davidebbo) [inline Kudu.Services/SiteExtensions/SiteExtensionController.cs]: No, that's fine, just making sure it wasn't some accidental code that wasn't meant to go in.
